dracut/10usr-generator/usr-generator

```diff
@@ -61,19 +79,25 @@ esac
 
 # Only proceed if the source is a path.
 if [[ "${usr}" != /* ]]; then
-  if [[ -z "${usr}" && -f /usr.squashfs ]]; then
+  # Here "$usr" does not contain the values of "mount.usr", hence the extra check not found in diskless-generator/ignition-generator
+  if [[ -z "${usr}" && -z "$(cmdline_arg mount.usr)" && -f /usr.squashfs ]]; then
     # Don't set "norecovery" for squashfs
     exit 0
   fi
   # Add "norecovery" to systemd-fstab-generator's unit and exit
   mount_usrflags=$(cmdline_arg mount.usrflags ro)
-  if [[ "$(echo "${mount_usrflags}" | grep -E '(^ro$|^ro,|,ro$|,ro,)')" != "" ]]; then
+  if [[ "$(echo "${mount_usrflags}" | grep -v -E '(^ro$|^ro,|,ro$|,ro,)')" != "" ]]; then
     # Don't set "norecovery" when mounting rw
     exit 0
   fi
   mount_usrflags="${mount_usrflags},norecovery"
-  mkdir -p "${UNIT_DIR}/sysroot-usr.mount.d"
-  cat >"${UNIT_DIR}/sysroot-usr.mount.d/10-norecovery.conf" <<EOF
+  # sysroot-usr.mount is a bind mount from /sysusr/usr
```

Comment:
Did you document how it always was, or /sysroot/usr being a bind mount from /sysusr/usr is your change? If the latter, I can't find the change in the PR.

Reply:
In this branch, `sysroot-usr.mount` and `sysusr-usr.mount` come from the upstream systemd generator, yes, it's how it always was and I didn't change it (but since it's a bind mount, adding ext4-specific options doesn't make sense and we need to do it for the real mount, hence the comment).